crypto/params.c

```diff
@@ -638,11 +645,14 @@ int OSSL_PARAM_set_BN(OSSL_PARAM *p, const BIGNUM *val)
 
     bytes = (size_t)BN_num_bytes(val);
     p->return_size = bytes;
+    if (sz < bytes)
+        return 0;
+
     if (p->data == NULL)
         return 1;
-    if (p->data_size >= bytes) {
-        p->return_size = p->data_size;
-        return BN_bn2nativepad(val, p->data, p->data_size) >= 0;
+    if (p->data_size >= sz) {
+        p->return_size = sz;
```

Comment:
And here also

Reply:
Here's a spot where I think that's taking purity too far.

Reply:
This is **exactly** where the _return_size_ **must** be set.

This is the number of bytes of BN that are being returned.  With the padding, the allocated space in the OSSL_PARAM might not be fully utilised and the get function needs to know this.  Witness the change to `OSSL_PARAM_get_BN()`.

Reply:
I agree that is the right spot. My comment was to what @slontis wrote

Reply:
Apologies, I thought the comment was directed at the change :(

Reply:
No prob, this is the usual one thread depth problem.  I could have specified it in my comment